### PR TITLE
Faang soumen branch

### PR DESCRIPTION
**Pull request by imsoumen**
# Have you read the [Contributing Guidelines](https://github.com/AkashSingh3031/The-Complete-FAANG-Preparation/blob/master/CONTRIBUTING.md)?
- [x] YES
- [ ] NO

## Project URL
- NA

## What is the current behavior?
- Issue Number: #... NA

## Category
- [ ] 𝑨𝑺𝑲 𝒀𝒐𝒖𝒓 𝑫𝒐𝒖𝒃𝒕𝒔
- [ ] 𝑴𝒊𝒔𝒄𝒆𝒍𝒍𝒂𝒏𝒆𝒐𝒖𝒔 𝑺𝒕𝒖𝒇𝒇.
- [ ] 𝑫𝑺𝑨
- [x] 𝑪𝒐𝒎𝒑𝒆𝒕𝒊𝒕𝒊𝒗𝒆 𝑷𝒓𝒐𝒈𝒓𝒂𝒎𝒎𝒊𝒏𝒈
- [ ] 𝑻𝒆𝒄𝒉𝒏𝒊𝒄𝒂𝒍 𝑺𝒖𝒃𝒋𝒆𝒄𝒕
- [ ] 𝑳𝒐𝒘 𝑳𝒆𝒗𝒆𝒍 𝑫𝒆𝒔𝒊𝒈𝒏
- [ ] 𝑷𝒓𝒐𝒋𝒆𝒄𝒕𝒔
- [ ] 𝑰𝒎𝒑𝒐𝒓𝒕𝒂𝒏𝒕 𝑩𝒐𝒐𝒌𝒔 𝒂𝒏𝒅 𝑹𝒆𝒔𝒐𝒖𝒓𝒄𝒆𝒔
- [ ] 𝑩𝒆𝒉𝒂𝒗𝒊𝒐𝒖𝒓𝒂𝒍 𝑰𝒏𝒕𝒆𝒓𝒗𝒊𝒆𝒘 𝑸𝒖𝒆𝒔𝒕𝒊𝒐𝒏𝒔

## Changes
- Description of changes
- Added Scala and Bash Solutions for 30 Days of Code

## **Screenshots** 
- (Add Running Work Demo Image)
- NA

## Checklist
- [x] I've read the [Contributing Guidelines](https://github.com/AkashSingh3031/The-Complete-FAANG-Preparation/blob/master/CONTRIBUTING.md).
- [x] I've checked the issue list before deciding what to submit.
- [x] I've follow [Tree of Index](https://github.com/AkashSingh3031/The-Complete-FAANG-Preparation#tree-of-index).
- [x] I've add work in proper folder
- [x] I've not delete/modify previous codes.

### CODE REVIEW

File: 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Scala/Day_10_Binary_Numbers.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
```
Comment (AkashSingh3031) [inline 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Scala/Day_10_Binary_Numbers.scala:1]: check this again

File: 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Bash/Day_8_Dictionaries_and_Map.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+Query 1: edward
+Edward is not one of the keys in our dictionary, so we print Not found.
+
+Query 2: harry
+Harry is one of the keys in our dictionary, so we print harry=12299933.
+
+"""
+
+sed 1d | awk 'NF==2{b[$1]=$2}NF==1{print b[$1] ? $1 "=" b[$1] : "Not found"}'
```
Comment (AkashSingh3031) [inline 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Bash/Day_8_Dictionaries_and_Map.sh:56]: check this again

File: 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Scala/Day_10_Binary_Numbers.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+Sample Case 1:
+The binary representation of 5 base-10 is 101 base-2 , so the maximum number of consecutive 1's is 1.
+
+Sample Case 2:
+The binary representation of 13 base-10 is 1101 base-2, so the maximum number of consecutive 1's is 2.
+*/
+
+(Scala Solution)
+
+object Solution {
```
Comment (AkashSingh3031) [inline 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Scala/Day_10_Binary_Numbers.scala:48]: check this again

**Review by AkashSingh3031 — COMMENTED** [review]
check `scala Q-10` & `BASH Q-8`

**Comment by imsoumen** [comment]
> check `scala Q-10` & `BASH Q-8`

Code changes done... Check now

File: 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Scala/Day_10_Binary_Numbers.scala (hunk 3 above)
Comment (imsoumen) [reply 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Scala/Day_10_Binary_Numbers.scala:48]: Done

File: 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Bash/Day_8_Dictionaries_and_Map.sh (hunk 2 above)
Comment (imsoumen) [reply 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Bash/Day_8_Dictionaries_and_Map.sh:56]: done

Comment (AkashSingh3031) [reply 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Bash/Day_8_Dictionaries_and_Map.sh:56]: > done

check 
image

File: 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Scala/Day_10_Binary_Numbers.scala (hunk 3 above)
Comment (AkashSingh3031) [reply 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Scala/Day_10_Binary_Numbers.scala:48]: > Done

check
image

File: 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Bash/Day_8_Dictionaries_and_Map.sh (hunk 2 above)
Comment (imsoumen) [reply 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Bash/Day_8_Dictionaries_and_Map.sh]: Removed this file for now.. Will upload later... seems like there is some problem with the test cases... It worked last year

File: 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Scala/Day_10_Binary_Numbers.scala (hunk 3 above)
Comment (imsoumen) [reply 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Scala/Day_10_Binary_Numbers.scala:68]: the file only had the actual code. now added the code with imports

Comment (imsoumen) [reply 3]. Competitive Programming/03]. HackerRank/1]. Practice/08]. 30 Days of Code/Scala/Day_10_Binary_Numbers.scala:68]: image

**Review by AkashSingh3031 — COMMENTED** [review]
Accepted

**Pull request merged**